### PR TITLE
mgr/dashboard_v2: Remove mock classes, tests are run against vstart cluster now

### PR DESCRIPTION
**Pull request by rjfd**
This PR changes the way unit tests are run and implemented. We were using mock classes to mock the mgr internal classes, this would prevent our unit tests to catch any changes made to the Ceph code.

Now, we removed the mock classes, adapted the unit tests code, and run the unit tests against a Ceph cluster created by the vstart script.

The tests are still run by `tox`.

Signed-off-by: Ricardo Dias <rdias@suse.com>

### CODE REVIEW

File: src/pybind/mgr/dashboard_v2/tests/test_ping.py [hunk 1]
```diff
@@ -8,20 +8,16 @@
 
 
 class PingTest(ControllerTestCase):
```
Comment (sebastian-philipp) [inline src/pybind/mgr/dashboard_v2/tests/test_ping.py]: Do we still need `PingTest` at all? I'd remove all echo and ping controllers now.

File: src/pybind/mgr/dashboard_v2/tests/test_rgw.py [hunk 2]
```diff
@@ -1,70 +1,39 @@
...
+        data = data[0]
+        self.assertIn('id', data)
+        self.assertIn('version', data)
+        self.assertIn('server_hostname', data)
+        self.assertIn('service', data)
+        self.assertIn('server', data)
+        self.assertIn('metadata', data)
+        self.assertIn('status', data)
+        self.assertIn('url', data)
+        self.assertIn('zone_name', data['metadata'])
```
Comment (sebastian-philipp) [inline src/pybind/mgr/dashboard_v2/tests/test_rgw.py:24]: This seems like a perfect candidate for [_ResponseValidator](https://bitbucket.org/openattic/openattic/src/660f1bb4979c14ed8023853e39b09362ef5cbb03/backend/rest_client.py?at=master&fileviewer=file-view-default#rest_client.py-49). Something for the future.

File: src/pybind/mgr/dashboard_v2/tests/test_rgw.py [hunk 3]
```diff
@@ -1,70 +1,39 @@
...
+        self.assertEqual('rgw', data['service']['type'])
+        self.assertEqual('/api/rgw/daemon/{}'.format(data['id']), data['url'])
 
-        cherrypy.tree.mount(RgwDaemon(mgr_mock), "/api/test/rgw")
+    @authenticate
+    def test_rgw_daemon_get(self):
+        self._get('/api/rgw/daemon')
+        data = self.jsonBody()
+        self._get(data[0]['url'])
+        data = self.jsonBody()
```
Comment (sebastian-philipp) [inline src/pybind/mgr/dashboard_v2/tests/test_rgw.py]: Can we make `_get` to return the JSON body per default?

Comment (rjfd) [reply src/pybind/mgr/dashboard_v2/tests/test_rgw.py]: yes, we can :)

File: src/pybind/mgr/dashboard_v2/tests/test_ping.py (hunk 1 above)
Comment (rjfd) [reply src/pybind/mgr/dashboard_v2/tests/test_ping.py]: yes we can remove it

**Review by p-se — DISMISSED** [review]

**Review by sebastian-philipp — DISMISSED** [review]

**Review by callithea — DISMISSED** [review]
Thanks for taking care of this! :)

**Review by sebastian-philipp — DISMISSED** [review]

**Review by sebastian-philipp — APPROVED** [review]

**Review by p-se — APPROVED** [review]

**Pull request merged**